testing/unittests/esdltests.cpp

```diff
@@ -1844,6 +1870,172 @@ constexpr const char * result = R"!!(<soap:Envelope xmlns:soap="http://schemas.x
+          </Transform>
+        </config>)!!";
+
+        static constexpr const char * data = R"!!(<?xml version="1.0" encoding="UTF-8"?>
+        <root>
+          <insert>
+            <common_value>178</common_value>
+            <common_r8>1.2</common_r8>
+            <Row>
+              <name>selected1</name>
+              <bval>65</bval>
+              <boolval>1</boolval>
+              <r4>3.4</r4>
+              <d>aa55aa55</d>
+              <ddd>1234567.89</ddd>
+              <u1>Straße1</u1>
+              <u2>ᚠᛇᚻ᛫ᛒᛦᚦ᛫ᚠᚱᚩᚠᚢᚱ᛫ᚠᛁᚱᚪ᛫ᚷᛖᚻᚹᛦᛚᚳᚢᛗ</u2>
+              <dt>2019-02-01 12:59:59</dt>
+            </Row>
+            <Row>
+              <name>selected2</name>
+              <bval>65</bval>
+              <boolval>1</boolval>
+              <r4>4.5</r4>
+              <d>bb66bb66</d>
+              <ddd>1234567.89</ddd>
+              <u1>Straße3</u1>
+              <u2>Straße4</u2>
+              <dt>2019-02-01 13:59:59</dt>
+            </Row>
+            <Row>
+              <name>selected3</name>
+              <bval>65</bval>
+              <boolval>1</boolval>
+              <r4>5.6</r4>
+              <d>cc77cc77</d>
+              <ddd>1234567.89</ddd>
+              <u1>Straße5</u1>
+              <u2>色は匂へど 散りぬるを</u2>
+              <dt>2019-02-01 14:59:59</dt>
+            </Row>
+          </insert>
+          <cities>
+            <name>aeiou</name>
+            <name>aeou</name>
+            <name>aoui</name>
+            <name>ei</name>
+            <name>aaa</name>
+            <name>bbb</name>
+          </cities>
+          <read>
+            <name>selected1</name>
+            <name>selected3</name>
+          </read>
+        </root>
+        )!!";
+
+        static constexpr const char * input = R"!!(<?xml version="1.0" encoding="UTF-8"?>
+        <root>
+          <Person>
+            <FullName>
+              <First>Joe</First>
+            </FullName>
+          </Person>
+        </root>
+        )!!";
+
+        static constexpr const char * script = R"!!(<es:CustomRequestTransform xmlns:es="urn:hpcc:esdl:script" target="Person">
+          <es:variable name="secret" select="'mydb'"/>
+          <es:variable name="database" select="'classicmodels'"/>
+          <es:variable name="section" select="'sql'"/>
+          <es:mysql secret="$secret" database="$database" section="$section" name="drop">
+            <es:sql>DROP TABLE IF EXISTS tbl1;</es:sql>
+          </es:mysql>
+          <es:mysql secret="$secret" database="$database" section="$section" name="create">
+            <es:sql>CREATE TABLE tbl1 ( name VARCHAR(20), bval BIT(15), value INT, boolval TINYINT, r8 DOUBLE, r4 FLOAT, d BLOB, ddd DECIMAL(10,2), u1 VARCHAR(10), u2 VARCHAR(10), dt DATETIME );</es:sql>
+          </es:mysql>
+          <es:mysql select="getDataSection('whatever')/this/insert/Row" secret="$secret" database="$database" section="$section" name="insert_each_row" result-delimiter="'inserted'">
```

Comment:
I'm of two minds regarding the implication of the select attribute in this command. First off it is clean, compact and powerful. But secondly it brings up some questions or mild concerns:

1. Are we setting the precedent that a select attribute in a command implies that the command operates on all elements of the select result? Are there cases where we haven't done this but perhaps we should? For example, I'm thinking of the <copy-of> and <xml-http-post> commands.  I think it will important to have consistent handling.
2. Is our treatment of the _result_ of the select consistent? If it is not quite exactly a node-set should it be? Should we be able to traverse to the parent of Row and refer to those common values?

Reply:
1. Yes that is the precedent for select.  For your examples, copy-of does treat select the same way, it copies the entire node-set.  Post-XML doesn't use select, but I could see us adding that in the future exactly as done in this PR.  Variables and parameters also use the whole node set.  The only ones I see that behave differently are operations like set-value, append-to-value, store-value, set-log-option, set-log-profile, which all treat the select like a value rather than a node-set... which I think is ok because it's basically black or white, not a grey area.

2. So there are two uses of select.  A. setting a single value, or B. selecting for processing.  B should probably always process the entire set.

Reply:
In a sense copy-of is the node set version of set-value.  I've used it that way in my tests to populate nodes with children rather than single values.  I think we also need a move function.